backend/src/dao/TeamEventsDao.ts

```diff
@@ -1,107 +1,64 @@
 import { v4 as uuidv4 } from 'uuid';
-import { DBTeamEvent } from '../types/DataTypes';
-import { memberCollection, teamEventsCollection, bucket } from '../firebase';
+import { teamEventsCollection, bucket } from '../firebase';
 import { NotFoundError } from '../utils/errors';
-import { getMemberFromDocumentReference } from '../utils/memberUtil';
 
 export default class TeamEventsDao {
-  static async getAllTeamEvents(): Promise<TeamEvent[]> {
+  static async getAllTeamEvents(): Promise<TeamEventInfo[]> {
     const eventRefs = await teamEventsCollection.get();
     return Promise.all(
       eventRefs.docs.map(async (eventRef) => {
-        const { name, date, numCredits, hasHours, requests, attendees, uuid, isCommunity } =
-          eventRef.data();
+        const { name, date, numCredits, hasHours, uuid, isCommunity } = eventRef.data();
         return {
           name,
           date,
           numCredits,
           hasHours,
-          requests: (await Promise.all(
-            requests.map(async (ref) => ({
-              ...ref,
-              member: await getMemberFromDocumentReference(ref.member)
-            }))
-          )) as TeamEventAttendance[],
-          attendees: (await Promise.all(
-            attendees.map(async (ref) => ({
-              ...ref,
-              member: await getMemberFromDocumentReference(ref.member)
-            }))
-          )) as TeamEventAttendance[],
           uuid,
           isCommunity
         };
       })
     );
   }
 
-  static async getTeamEvent(uuid: string): Promise<TeamEvent> {
+  static async getTeamEvent(uuid: string): Promise<TeamEventInfo> {
     const eventDoc = teamEventsCollection.doc(uuid);
     const eventRef = await eventDoc.get();
     if (!eventRef.exists) throw new NotFoundError(`No form with uuid '${uuid}' found.`);
     const eventForm = eventRef.data();
     if (eventForm == null)
       throw new NotFoundError(`No form content in form with uuid '${uuid}' found.`);
     return {
-      ...eventForm,
-      requests: (await Promise.all(
-        eventForm.requests.map(async (ref) => ({
-          ...ref,
-          member: await getMemberFromDocumentReference(ref.member)
-        }))
-      )) as TeamEventAttendance[],
-      attendees: (await Promise.all(
-        eventForm.attendees.map(async (ref) => ({
-          ...ref,
-          member: await getMemberFromDocumentReference(ref.member)
-        }))
-      )) as TeamEventAttendance[]
+      ...eventForm
```

Comment:
Rewrite as `return eventForm`. There is no need to use the spread operator here.